lib/sshutils/sftp/sftp_test.go

```diff
@@ -204,6 +247,16 @@ func TestUpload(t *testing.T) {
 			} else {
 				require.EqualError(t, err, fmt.Sprintf(tt.expectedErr, tempDir))
 			}
+
+			if tt.tryOverwrite {
```

Comment:
I don't follow why we need another block, nor why the above doesn't error. Could the cfg.transfer above run this scenario?

Reply:
Agree with @codingllama here - this block seems to just repeat the same test without changing anything, and I am confused how the tests pass when `tt.expectedErr != tt.expectedOverwriteErr` - it looks like the first time the tests run, it should fail the test as `tt.expectedErr == ""`.

Reply:
Mind adding a comment? Doesn't have to be long, but having something would help.

Reply:
Thanks. I saw that the file mode was specified as 0o440, but it didn't click that it was in the src location and that you first needed to copy it with that mode to the destination, and then do it again to actually test the new logic.